server/game/GameActions/ReturnToDeckAction.js

```diff
@@ -34,7 +34,7 @@ class ReturnToDeckAction extends CardGameAction {
                 event.context.refillProvince(card.controller, card.location, context);
             }
             card.owner.moveCard(card, destination, { bottom: this.bottom });
-            if(this.shuffle) {
+            if(this.shuffle && (card === this.target[this.target.length - 1])) {
```

Comment:
This won't work if getEvent is called outside of the context of an ability (which happens in some places I think)

Reply:
yes, that's fine